Jenkinsfile

```diff
@@ -0,0 +1,112 @@
+pipeline {
+  agent none
+  options{
+    skipDefaultCheckout()
+  }
+  stages {
+    stage('Build') {
+      agent{
+        label 'noida-linux-ubuntu16-ci-slave'
+      }
+      steps {
+        checkout scm
+        withEnv(["GOROOT=/home/jenkins/go", "GOPATH=/home/jenkins/goprojects", "PATH+GO=/home/jenkins/goprojects/bin:/home/jenkins/go/bin"]){
+          withCredentials([usernamePassword(credentialsId: 'ldap_credentials', passwordVariable: 'PASSWORD', usernameVariable: 'USERNAME')]) {
+            sh "cd $HOME/goprojects/src/github.com/armPelionEdge && if [ ! -d \"maestro\" ]; then git clone https://github.com/sameer2209-arm/maestro.git && git checkout ${env.BRANCH_NAME} && go build; else cd maestro && git checkout ${env.BRANCH_NAME} && git pull origin ${env.BRANCH_NAME}; fi"
+            sh "cd $HOME/goprojects/src/github.com/armPelionEdge && if [ ! -d \"greasego\" ]; then git clone https://github.com/armPelionEdge/greasego.git; else cd greasego && git pull; fi"
+            sh 'cd /home/jenkins/goprojects/src/github.com/armPelionEdge/greasego && ./build-deps.sh && ./build.sh'
+            //sh 'cd $HOME/goprojects/src/github.com/armPelionEdge/maestro && export GOBIN=$GOPATH/bin && export PATH=$GOBIN:$PATH && ./build-deps.sh && touch maestro.config && DEBUG=1 DEBUG2=1 ./build.sh && LD_LIBRARY_PATH=vendor/github.com/armPelionEdge/greasego/deps/lib maestro -config test-ok-string.config'
+            sh 'cd /home/jenkins/goprojects/src/github.com/armPelionEdge/maestro && export GOBIN=$GOPATH/bin && export PATH=$GOBIN:$PATH && ./build-deps.sh && touch maestro.config && DEBUG=1 DEBUG2=1 ./build.sh'
+          }
+        }
+      }
+    }
+    
+    /*stage('Test and Code Review') {
+      parallel {*/
+        stage('Test'){
```

Comment:
Might be nice to add the little black box tests we run here: https://austin-gw-jenkins.arm.com/job/maestro/configure